src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -380,8 +424,10 @@ public static OptionData fromData(@Nonnull DataObject json)
                 choices1.stream(DataArray::getObject).forEach(o ->
                 {
                     Object value = o.get("value");
-                    if (value instanceof Number)
+                    if (value instanceof Long)
```

Comment:
There is no way jackons will parse all integers into longs

Reply:
So return back to Number?

Reply:
I've gone ahead and used the `if (o.isType(...))` setup you suggested, because it feels the cleanest to me compared to an instanceof check.